app/src/main/java/com/example/farmeraid/data/FarmRepository.kt

```diff
@@ -0,0 +1,68 @@
+package com.example.farmeraid.data
+
+import android.util.Log
+import com.example.farmeraid.data.model.FarmModel
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.FieldValue
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.coroutines.tasks.await
+
+class FarmRepository(
+    private val userRepository: UserRepository
+) {
+    private var db: FirebaseFirestore = FirebaseFirestore.getInstance()
+
+    suspend fun createFarm(farmName: String){
+        //if db.collection("farm").get
+        try {
+            userRepository.getUserId()?.let{
+                val docRef = db.collection("farm").add(
+                    mapOf (
+                        "name" to farmName,
+                        "users" to listOf(it),
+                        "markets" to emptyList<String>(),
+                        "charities" to emptyList<String>()
+                    )
+                ).await()
+
+                db.collection("inventory").document(docRef.id).set(
+                    mapOf("produce" to emptyMap<String,Int>())
+                )
+
+                //TODO create new transaction document for the farm
+
+                //TODO update userRespository with new farmID
```

Comment:
We would also set the user field as admin, as they created the farm

Reply:
I believe we need a function to set the User's farmId which we would call here